### ISSUE TITLE
Undocumented BC break - Longo ption must be longer than 1 char

### ISSUE THREAD

**Issue by JanMikes** [getopt-php/getopt-php#143]
Hello, i have just upgraded from 1.1 to latest version and i receive this exception:  
`Long option must be null or an alphanumeric string, found 'c'`

We have this option: `[null, 'c', GetOpt::OPTIONAL_ARGUMENT, 'Set working copy.'],`

I have found out the issue is in `Option::setLong()` where this regex wont pass: `/^[a-zA-Z0-9][a-zA-Z0-9_-]{1,}$/` - https://regex101.com/r/v3F9pu/1

It worked fine with version 1.1, but now it does not.

If i change the regex like this, it works: `/^[a-zA-Z0-9][a-zA-Z0-9_-]*$/` https://regex101.com/r/qKbrCc/2

The problem is it expects at least 2 charactes.

**Comment by JanMikes** [comment]
In test `test/Options/CommonTest.php` i found out this comment:
```php
[ null, 'a', GetOpt::NO_ARGUMENT ]        // long name must be at least 2 characters long
```

Why is it like this? We have been using long 1 char option for years :-)

**Comment by tflori** [comment]
oh oh..

yes, basically we could allow `--a` as option. maybe just add it again. If no one votes against it I'm free for a MR.

**Comment by tflori** [comment]
solved with #144
